devices/Hdc1080/Hdc1080.cs

```diff
@@ -0,0 +1,183 @@
+
+        /// <summary>
+        /// Constructs Hdc1080 instance with high temperature and humidity sampling. Measurement mode set by default to humidity and temperature.
+        /// </summary>
+        /// <param name="i2cDevice">I2C device used to communicate with the device</param>
+        /// <param name="humidityResolution">Humidity sampling resolution</param>
+        /// <param name="temperatureResolution">Temperature sampling resolution</param>
+        public Hdc1080(I2cDevice i2cDevice, HumidityResolution humidityResolution = HumidityResolution.High, TemperatureResolution temperatureResolution = TemperatureResolution.High)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            // According to data sheet sensor needs at least 15ms after wake up to be ready.
+            Thread.Sleep(_wakeUpTimeMilliseconds);
+            SetTemperatureResolution(temperatureResolution);
+            SetHumidityResolution(humidityResolution);
+        }
+
+        /// <summary>
+        /// Reads temperature, check data sheet, page 14, 8.6.1 section
+        /// </summary>
+        /// <returns>
+        /// Temperature
+        /// </returns>
+        [Telemetry("Temperature")]
+        public Temperature ReadTemperature() => Temperature.FromDegreesCelsius((ReadData(RegisterAddress.TemperatureMeasurement) * 165 / 65536f) - 40);
+
+        /// <summary>
+        /// Reads relative humidity, check data sheet, page 14, 8.6.2 section
+        /// </summary>
+        /// <returns>
+        /// Relative humidity
+        /// </returns>
+        [Telemetry("Humidity")]
+        public RelativeHumidity ReadHumidity() => RelativeHumidity.FromPercent(ReadData(RegisterAddress.HumidityMeasurement) * 100 / 65536);
+
+        /// <summary>
+        /// Reads device id, check data sheet, page 14, 8.6.6 section
+        /// </summary>
+        /// <returns>
+        /// Device id
+        /// </returns>
+        public string ReadDeviceId() => ReadData(RegisterAddress.DeviceId).ToString();
+
+        /// <summary>
+        /// Reads manufacturer id, check data sheet, page 14, 8.6.5 section
+        /// </summary>
+        /// <returns>
+        /// Manufacturer id
+        /// </returns>
+        public string ReadManufacturerId() => ReadData(RegisterAddress.ManufacturerId).ToString();
+
+        /// <summary>
+        /// Reads serial number, check data sheet, page 14, 8.6.4 section
+        /// </summary>
+        /// <returns>
+        /// Serial number
+        /// </returns>
+        public string ReadSerialNumber() => $"{ReadData(RegisterAddress.SerialIdFirstByte)}{ReadData(RegisterAddress.SerialIdSecondByte)}{ReadData(RegisterAddress.SerialIdThirdByte)}";
+
+        /// <summary>
+        /// Sets humidity resolution
+        /// </summary>
+        /// <param name="resolution">Humidity resolution</param>
+        public void SetHumidityResolution(HumidityResolution resolution)
```

Comment:
This can be a property as well where you can get and set the HumidyResolution. I feel it's more natural. Same for the TemperatureResolution. I guess there is a way as well to read it (or store in private property)